### ISSUE TITLE
F1210 08:16:15.441911 10686 layer_factory.hpp:80] Check failed: registry.count(type) == 1 (0 vs. 1) Unknown layer type: Python 

### ISSUE THREAD

**Issue by wowo200** [rbgirshick/py-faster-rcnn#38]
Hi!

After did the basic installation, when I ran the demo.py code, I got the check error.

 F1210 08:16:15.441911 10686 layer_factory.hpp:80] Check failed: registry.count(type) == 1 (0 vs. 1) Unknown layer type: Python (known types: AbsVal, Accuracy, ArgMax, BNLL, Concat, ContrastiveLoss, Convolution, Data, Deconvolution, Dropout, DummyData, Eltwise, Embed, EuclideanLoss, Exp, Filter, Flatten, HDF5Data, HDF5Output, HingeLoss, Im2col, ImageData, InfogainLoss, InnerProduct, LRN, Log, MVN, MemoryData, MultinomialLogisticLoss, PReLU, Pooling, Power, ROIPooling, ReLU, Reduction, Reshape, SPP, Sigmoid, SigmoidCrossEntropyLoss, Silence, Slice, SmoothL1Loss, Softmax, SoftmaxWithLoss, Split, TanH, Threshold, Tile, WindowData)

how to fix this?

**Comment by leejiajun** [comment]
I got that check error too.

**Comment by szi** [comment]
Same error on my machine.

**Comment by leejiajun** [comment]
@szi thank you, it was solved.

**Comment by yaochx** [comment]
@szi as you said, it was solved.

**Comment by ran5515** [comment]
Hello, I tried "WITH_PYTHON_LAYER := 1" and make clean and make again.But the error still exists.
I am using Mac Sierra.
Can anyone help me with it?

"
I0220 11:06:37.499789 3922367424 layer_factory.hpp:77] Creating layer proposal
F0220 11:06:37.499804 3922367424 layer_factory.hpp:81] Check failed: registry.count(type) == 1 (0 vs. 1) Unknown layer type: Python (known types: AbsVal, Accuracy, ArgMax, BNLL, BatchNorm, BatchReindex, Bias, Concat, ContrastiveLoss, Convolution, Data, Deconvolution, Dropout, DummyData, ELU, Eltwise, Embed, EuclideanLoss, Exp, Filter, Flatten, HDF5Data, HDF5Output, HingeLoss, Im2col, ImageData, InfogainLoss, InnerProduct, LRN, Log, MVN, MemoryData, MultinomialLogisticLoss, PReLU, Pooling, Power, ROIPooling, ReLU, Reduction, Reshape, SPP, Scale, Sigmoid, SigmoidCrossEntropyLoss, Silence, Slice, SmoothL1Loss, Softmax, SoftmaxWithLoss, Split, TanH, Threshold, Tile, WindowData)
*** Check failure stack trace: ***
"

**Comment by dksakkos** [comment]
@ran5515 I also tried "WITH_PYTHON_LAYER := 1" and I still get the error. Did you solve it?

**Comment by derkbreeze** [comment]
I tried WITH_PYTHON_LAYER := 1, but still facing the same error
although I successfully build the caffe and pycaffe.

**Comment by YIN0923** [comment]
> make clean，然后make to rebuid
how do this ?

**Comment by cqray1990** [comment]
> Just ran into same error, then came across this:
> [rbgirshick/fast-rcnn#31](https://github.com/rbgirshick/fast-rcnn/issues/31)
> 
> You have to uncomment WITH_PYTHON_LAYER := 1 in the Makefile.config, then make clean, then make to rebuild
i uncomment WITH_PYTHON_LAYER := 1 in the Makefile.config and remove build and sudo make -j4 all but is also get the same error

**Comment by cqray1990** [comment]
> Just ran into same error, then came across this:
> [rbgirshick/fast-rcnn#31](https://github.com/rbgirshick/fast-rcnn/issues/31)
> 
> You have to uncomment WITH_PYTHON_LAYER := 1 in the Makefile.config, then make clean, then make to rebuild
when  test an image it have no  errors,but when i train ,it get this errors:
F0220 11:06:37.499804 3922367424 layer_factory.hpp:81] Check failed: registry.count(type) == 1 (0 vs. 1) Unknown layer type: Python (known types: AbsVal, Accuracy, ArgMax, BNLL, BatchNorm, BatchReindex, Bias, Concat, ContrastiveLoss, Convolution, Data, Deconvolution, Dropout, DummyData, ELU, Eltwise, Embed, EuclideanLoss, Exp, Filter, Flatten, HDF5Data, HDF5Output, HingeLoss, Im2col, ImageData, InfogainLoss, InnerProduct, LRN, Log, MVN, MemoryData, MultinomialLogisticLoss, PReLU, Pooling, Power, ROIPooling, ReLU, Reduction, Reshape, SPP, Scale, Sigmoid, SigmoidCrossEntropyLoss, Silence, Slice, SmoothL1Loss, Softmax, SoftmaxWithLoss, Split, TanH, Threshold, Tile, WindowData)

**Comment by cqray1990** [comment]
@derkbreeze 
did you solver it?
F0609 12:08:19.237373 13764 layer_factory.hpp:81] Check failed: registry.count(type) == 1 (0 vs. 1) Unknown layer type: ImageLabelmapData (known types: AbsVal, Accuracy, ArgMax, BNLL, BatchNorm, BatchReindex, Bias, Clip, Concat, ContrastiveLoss, Convolution, Crop, Data, Deconvolution, Dropout, DummyData, ELU, Eltwise, Embed, EuclideanLoss, Exp, Filter, Flatten, HDF5Data, HDF5Output, HingeLoss, Im2col, ImageData, InfogainLoss, InnerProduct, Input, LRN, LSTM, LSTMUnit, Log, MVN, MemoryData, MultinomialLogisticLoss, PReLU, Parameter, Pooling, Power, Python, RNN, ReLU, Reduction, Reshape, SPP, Scale, Sigmoid, SigmoidCrossEntropyLoss, Silence, Slice, Softmax, SoftmaxWithLoss, Split, Swish, TanH, Threshold, Tile, WindowData)
